### PR TITLE
add standard deviation attribute

### PR DESCRIPTION
**Pull request by AdilKhurshid**
closes https://github.com/equinor/vds-slice/issues/127

### CODE REVIEW

**Review by achaikou — COMMENTED** [review]
Nice! :smile_cat:

One comment to the commit name though:
Can you maybe start it with upper case, keep the log consistent?

> Add standard deviation attribute

?

`+` hint: if you instead of `issue #127` type `closes #127`, github will automatically close the issue once PR is merged

**Review by achaikou — COMMENTED** [review]
Looks good! Prepare for merge: rebase and fix up commits.

Plus: can I get please feedback for each review comment :innocent: ?
It is of course easy to see here that you applied all the comments, but in general it would be cool to get a comment back or :+1: for everything you agree with and implemented, so I know what I am supposed to expect :)

**Comment by AdilKhurshid** [comment]
> Looks good! Prepare for merge: rebase and fix up commits.
> 
> Plus: can I get please feedback for each review comment 😇 ? It is of course easy to see here that you applied all the comments, but in general it would be cool to get a comment back or 👍 for everything you agree with and implemented, so I know what I am supposed to expect :)

Thanks for reviewing the pull request and I'm extremely sorry I didn't add the back comments. I actually messed up my git commit and was trying to fix that I totally forgot to reply.

File: internal/vds/attribute.cpp [hunk 1]
```diff
@@ -77,4 +77,20 @@ void rms(Horizon const& horizon, void* dst, std::size_t size) noexcept (false) {
...
+    std::size_t vsize = horizon.vsize();
+
+    auto sdfunc = [vsize](Horizon::VerticalIt beg, Horizon::VerticalIt end) {
+        float sum = std::accumulate(beg, end, 0.0f);
+        float stdSum = std::accumulate(beg, end, 0,
+        [&](float a, float b){
+            return a + std::pow((b-(sum / vsize)),2) ;
+            }
+        );
+        return sqrt(stdSum / vsize);
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply internal/vds/attribute.cpp]: Added suggested changes.

File: internal/vds/attribute.cpp [hunk 2]
```diff
@@ -77,4 +77,20 @@ void rms(Horizon const& horizon, void* dst, std::size_t size) noexcept (false) {
...
 }
 
+void sd(Horizon const& horizon, void* dst, std::size_t size) noexcept (false) {
+    std::size_t vsize = horizon.vsize();
+
+    auto sdfunc = [vsize](Horizon::VerticalIt beg, Horizon::VerticalIt end) {
+        float sum = std::accumulate(beg, end, 0.0f);
+        float stdSum = std::accumulate(beg, end, 0,
+        [&](float a, float b){
+            return a + std::pow((b-(sum / vsize)),2) ;
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply internal/vds/attribute.cpp]: Storing the calculated mean in a separate variable as you suggested to remove unnecessary operations.

fixed indentation

**Comment by AdilKhurshid** [comment]
> Looks good! Prepare for merge: rebase and fix up commits.
> 
> Plus: can I get please feedback for each review comment 😇 ? It is of course easy to see here that you applied all the comments, but in general it would be cool to get a comment back or 👍 for everything you agree with and implemented, so I know what I am supposed to expect :)

Thanks for reviewing the pull request and I'm extremely sorry I didn't add the back comments. I actually missed my git commit and was trying to fix that I totally forgot that.

**Review by achaikou — APPROVED** [review]
Don't worry, absolutely no reason to be sorry!

I am just preparing you for future larger reviews where it would be very useful to know if you agree and intend to implement the comment, or disagree, or agree but don't want to fix it, or something else :)

We here usually go with :+1: for simple "yes agree would fix" as replying to majority of comments with equivalent of "Done" and "Fixed" or so is just boooring.

But looks great, approved!

**Pull request merged**